### PR TITLE
improved stepper that walks using Google Directions API

### PR DESCRIPTION
**Pull request by th3w4y**
Short Description: 
In order to try it, add below lines to pokecli.py

- from pokemongo_bot.polyline_stepper import PolylineStepper 
- bot.stepper=PolylineStepper(bot) [after the bot.start()]

Fixes:
- fixed percentage_walked calculation
- fixed some bugs in polyline_stepper
- changed speed to 100m/s inside the test 
- merged with current (dev) - some conflicts solved

### CODE REVIEW

**Comment by TheSavior** [comment]
This is great work. 

It would make things much easier if your PR actually described the problem you are trying to solve and your changes. Having many PRs that just say Polyline isn't very helpful to reviewers

**Comment by jtdroste** [comment]
Cloud you rebase your commits, and merge them into one commit?

**Comment by th3w4y** [comment]
@jtdroste Thanks for the suggestion - i just learned how to do it now

File: pokemongo_bot/polyline_stepper.py [hunk 1]
```diff
@@ -1,56 +1,39 @@
...
+        origin = ','.join([str(i2f(self.api._position_lat)), str(i2f(self.api._position_lng))])
         destination = ','.join([str(lat), str(lng)])
-        polyline_walker = PolylineWalker(origin, destination, self.speed)
+        polyline_walker = PolylineWalker(origin, destination, speed)
         proposed_origin = polyline_walker.points[0]
         proposed_destination = polyline_walker.points[-1]
         proposed_lat = proposed_origin[0]
         proposed_lng = proposed_origin[1]
         if proposed_lat != lat and proposed_lng != lng:
+            logger.log('[#] Using _old_walk_to to go to the proposed_origin: ' +str(proposed_origin))
```
Comment (ProjectBarks) [inline pokemongo_bot/polyline_stepper.py]: please use string format

**Comment by solderzzc** [comment]
Test code, not called, so merged.

**Pull request merged**